rfd/0208-proxy-desktop-session.md

```diff
@@ -0,0 +1,112 @@
+
+### Session Initialization Flow
+
+1. The connection is initiated by tsh, which opens the `ProxyDesktopSession` gRPC stream.
+   * The first message sent by tsh must contain the `dial_target`, specifying the desktop name and target cluster. 
+   * If the first message does not include the `dial_target`, the Proxy will immediately terminate the connection.
+2. The Teleport Proxy uses the provided `dial_target` to dial the Desktop Service, following the logic in a Web UI handler.
+   * The Proxy queries the cluster's desktops looking for the ones that match the `desktop_name` in `dial_target`.
+   * The returned desktops are then mapped to desktop services reporting them.
+   * The Proxy attempts to connect to these services in a randomized order until one succeeds.
+3. Data tunneling starts.
+   * Once a Desktop Service is successfully reached, the Proxy begins forwarding raw data between tsh and the Desktop Service.
+   * Meanwhile, tsh initiates a TLS connection over the gRPC stream using its MFA-verified client certificate.
+   * The TLS connection is established when the Desktop Service accepts the handshake.
+
+Note that Proxy won't enforce RBAC checks. It will attempt to connect to the Windows Desktop Service even if the user doesn’t have access to the target desktop.
+The Desktop Service will enforce access control and return an error if the user isn’t allowed.
+
+### Drawbacks and considered alternatives
+
+Tunneling a TLS connection over a gRPC stream introduces some overhead due to the added TLS metadata in each message. 
+This could impact performance slightly, although manual testing did not show any noticeable latency.
+
+A potential alternative design would terminate the client's TLS connection in the Proxy, then open a separate TLS connection to the Desktop Service. 
+This would require the Proxy to detect whether per-session MFA is needed, issue an MFA challenge to the client and verify the response.
+However, this approach is not currently feasible, since Proxy has no straightforward way to determine whether per-session MFA is required 
+for a given user—it doesn’t interact with the Auth Service using a user's identity.
+
+### gRPC Definitions
+
+#### Service definition
+
+```protobuf
+// Bidirectional stream for proxying desktop sessions.
+rpc ProxyDesktopSession(stream ProxyDesktopSessionRequest) returns (stream ProxyDesktopSessionResponse);
+```
+
+#### Message definitions
+
+```protobuf
+// Request message for a proxied desktop session.
+message ProxyDesktopSessionRequest {
+  // Target cluster and desktop. Must be sent as the first message.
+  TargetDesktop dial_target = 1;
+
+   // Payload from Teleport Desktop Protocol. 
+   DesktopSessionMessage message = 2;
+}
+
+// Response message for a proxied desktop session.
+message ProxyDesktopSessionResponse {
+   // Payload from Teleport Desktop Protocol.
+   DesktopSessionMessage message = 1;
+}
```

Comment:
Do we need any info about the cluster like we do in SSH, like whether or not we're FIPS?

Reply:
AFAIK we don't need such info.